File: start.rb
require 'open-uri'
require 'nokogiri'
require 'enumerator'
require 'fileutils'

url = "https://gobyexample.com"
html = open(url)
doc = Nokogiri::HTML(html)
keywords = doc.xpath("//li//a/@href")
basedir = ARGV[0] || "learn-go-by-example"

Dir.mkdir basedir

keywords = keywords.enum_for(:each_with_index).map do |keyword, index|
  filename = "%02d-#{keyword}.go" % index
  path = "#{basedir}/%02d-#{keyword}" % index
  FileUtils::mkdir_p path
  File.new("#{path}/#{filename}", "w")
end

File: start.py
from lxml import html
import requests
import os
import sys

page = requests.get('https://gobyexample.com/')
tree = html.fromstring(page.content)
keywords = tree.xpath('//li//a/@href')

basedir = sys.argv[1] or 'learn-go-by-example'

def create_structure(dir):
    for index, keyword in enumerate(keywords):
        directory = '{}/{:02d}-{}'.format(dir, index, keyword)
        filename = '{:02d}-{}.go'.format(index, keyword)
        path  = '{}/{}'.format(directory, filename)
        os.makedirs(directory)
        if not os.path.exists(path):
            open(path, 'w').close()


if not os.path.exists(basedir):
    os.makedirs(basedir)
    create_structure(basedir)
else:
    print 'Directory already exists: {}'.format(basedir)
